src/timeline/InputBar.cpp

```diff
@@ -249,6 +250,54 @@ InputBar::updateAtRoom(const QString &t)
         this->containsAtRoom_ = roomMention;
         emit containsAtRoomChanged();
     }
+
+    // check for invalid commands
+    auto commandName = getCommandAndArgs().first;
+    bool hasInvalidCommand{};
+    if (!commandName.isNull() && '/' + commandName != text()) {
+        static const QStringList validCommands{QStringLiteral("me"),
+                                               QStringLiteral("react"),
+                                               QStringLiteral("join"),
+                                               QStringLiteral("knock"),
+                                               QStringLiteral("part"),
+                                               QStringLiteral("leave"),
+                                               QStringLiteral("invite"),
+                                               QStringLiteral("kick"),
+                                               QStringLiteral("ban"),
+                                               QStringLiteral("unban"),
+                                               QStringLiteral("redact"),
+                                               QStringLiteral("roomnick"),
+                                               QStringLiteral("shrug"),
+                                               QStringLiteral("fliptable"),
+                                               QStringLiteral("unfliptable"),
+                                               QStringLiteral("sovietflip"),
+                                               QStringLiteral("clear-timeline"),
+                                               QStringLiteral("reset-state"),
+                                               QStringLiteral("rotate-megolm-session"),
+                                               QStringLiteral("md"),
+                                               QStringLiteral("cmark"),
+                                               QStringLiteral("plain"),
+                                               QStringLiteral("rainbow"),
+                                               QStringLiteral("rainbowme"),
+                                               QStringLiteral("notice"),
+                                               QStringLiteral("rainbownotice"),
+                                               QStringLiteral("confetti"),
+                                               QStringLiteral("rainbowconfetti"),
+                                               QStringLiteral("goto"),
+                                               QStringLiteral("converttodm"),
+                                               QStringLiteral("converttoroom")};
+        hasInvalidCommand = !validCommands.contains(commandName);
```

Comment:
A std::set, std::unordered_set, QHash, etc would probably be slightly more efficient.